src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -195,7 +195,7 @@ bool IHttpBodyControlFeature.AllowSynchronousIO
         set => AllowSynchronousIO = value;
     }
 
-    bool IHttpMaxRequestBodySizeFeature.IsReadOnly => HasStartedConsumingRequestBody || IsUpgraded;
+    bool IHttpMaxRequestBodySizeFeature.IsReadOnly => HasStartedConsumingRequestBody || IsUpgraded || IsExtendedConnectRequest;
```

Comment:
@halter73 this is what you recommended, but it's actually stricter than Upgrade which only prevents modification after upgrade is called. Neither null out the size limit field, they just ignore it. The data rate limit property is similarly ignored.

Reply:
I think this is good. With extended CONNECT requests, middleware has not chance to run before the "upgrade" so-to-speak. The request is pre-upgraded.